pkg/restore/db.go

```diff
@@ -37,6 +44,23 @@ func NewDB(g glue.Glue, store kv.Storage) (*DB, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	return se, nil
+}
+
+func (db *DB) contextualSession(ctx context.Context) glue.Session {
+	if session, ok := ctx.Value(SessionInContext).(glue.Session); ok {
+		return session
+	}
+	return db.se
+}
```

Comment:
ok i don't understand why the change is at the Session level. why create a pool of Session rather than a pool of DB?

Reply:
Just because this is the method to implement isolated sessions that I figured out :|

You're right.